examples/self-diagnostics/src/main.rs

```diff
@@ -0,0 +1,93 @@
+use opentelemetry::global::{self, set_error_handler, Error as OtelError};
+use opentelemetry::KeyValue;
+use opentelemetry_appender_tracing::layer;
+use opentelemetry_sdk::logs::LoggerProvider;
+use opentelemetry_sdk::metrics::{PeriodicReader, SdkMeterProvider};
+use opentelemetry_sdk::{runtime, Resource};
+use tracing_subscriber::prelude::*;
+use tracing_subscriber::EnvFilter;
+
+use std::error::Error;
+use tracing::error;
+
+fn custom_error_handler(err: OtelError) {
+    match err {
+        OtelError::Metric(err) => error!("OpenTelemetry metrics error occurred: {}", err),
+        OtelError::Trace(err) => error!("OpenTelemetry trace error occurred: {}", err),
+        OtelError::Log(err) => error!("OpenTelemetry log error occurred: {}", err),
+        OtelError::Propagation(err) => error!("OpenTelemetry propagation error occurred: {}", err),
+        OtelError::Other(err_msg) => error!("OpenTelemetry error occurred: {}", err_msg),
+        _ => error!("An unknown OpenTelemetry error occurred"), //won't reach here
+    }
+}
```

Comment:
This is pretty much the same as the default handler but using `traing::error!` macros?

Reply:
yes true. This is to demonstrate using OpenTelemetry logging pipeline as custom handler for self-diagnostics.